astropy/cosmology/core.py

```diff
@@ -165,22 +353,98 @@ def __eq__(self, other):
         Returns
         -------
         bool
-            True if all immutable fields are equal, False otherwise.
+            True if cosmologies are equivalent, False otherwise.
         """
+        # The options are: 1) same class & parameters; 2) same class, different
+        # parameters; 3) different classes, equivalent parameters; 4) different
+        # classes, different parameters. (1) & (3) => True, (2) & (4) => False.
+
         if not isinstance(other, Cosmology):
             return False
```

Comment:
SGTM. It would be very cool to allow equality and equivalence checks against 3rd-party cosmologies using ``to_format()``. I made a card for this. https://github.com/astropy/astropy/projects/7#card-69385769

Reply:
Essentially if it fails the class check it will try `other = Cosmology.from_format(other)` and then proceed